library/Rules/BankAccount.php

```diff
@@ -0,0 +1,49 @@
+<?php
+namespace Respect\Validation\Rules;
+
+use malkusch\bav\BAV;
+use Respect\Validation\Exceptions\ComponentException;
+
+/**
+ * Validates a bank account for a given bank.
+ *
+ * Currently only German validation is supported (country code "de").
+ * This validator depends on the composer package malkusch/bav.
+ *
+ * @author Markus Malkusch <markus@malkusch.de>
+ * @see BAV::isValidBankAccount()
+ */
+class BankAccount extends Callback
+{
+
+    /**
+     * Sets the country code and bank.
+     *
+     * The country code is not case sensitive.
+     *
+     * @param string $countryCode The ISO 639-1 country code.
+     * @param string $bank The bank.
+     */
+    public function __construct($countryCode, $bank)
+    {
+        $callback = null;
+
+        switch (strtolower($countryCode)) {
+            case "de":
+                $bav = new BAV();
+                $callback = function ($account) use ($bank, $bav) {
+                    return $bav->isValidBankAccount($bank, $account);
+                };
```

Comment:
Because this is the place where the array callback doesn't work anymore. The parameter order wouldn't match for `isValidBankAccount($bank, $account)`. `Callback::validate()` would call `isValidBankAccount($account, $bank)`. Even if the `array_unshift()` would be a `array_push()` it's very bad practice to rely on that implementation detail.

So the best solution here was the anonymous function. As you Respect is PHP>=5.3 there is no problem using them. I used them in the other places because of consistency.